.changeset/unlucky-ducks-raise.md

```diff
@@ -0,0 +1,111 @@
+---
+'@keystone-6/core': major
+---
+#### Breaking
+
+##### Move of `image` and `files` in the keystone config into new option `storage`
+
+The `image` and `files` config options have been removed from Keystone's config - the configuration has
+been moved into a new `storage` configuration object.
+
+Old:
+
+```ts
+export default config({
+    image: { upload: 's3' },
+    lists: { 
+        images: { fields: { image() }
+        /* ... */
+    },
+    /* ... */
+})
+```
+
+New:
+
+```ts
+export default config({
+    storage: {
+        my_image_storage: {
+            kind: 's3',
+            type: 'image',
+            bucketName: S3_BUCKET_NAME,
+            region: S3_REGION,
+            accessKeyId: S3_ACCESS_KEY_ID,
+            secretAccessKey: S3_SECRET_ACCESS_KEY
+        },
+    }
+    lists: {
+        images: { fields: image({ storage: 'my_image_storage' }) }
+        /* ... */
+    },
+    /* ... */
+})
+```
+
+##### Removal of refs for `images` and `files`
+
+Refs were an interesting situation! They allowed you to link images stored in your storage source (s3 or local), and use the same
+image anywhere else images are used. This causes a bunch of complication, and prevented Keystone ever reliably being able
+to remove images from your source, as it couldn't easily track where images were used.
+
+To simplify things for you and us, we're removing `refs` as a concept, but don't panic just yet, we are conceptually replacing
+them with something you are already familiar with: `relationships`.
+
+If you wanted refs, where images could be available in multiple places, our new recommendation is:
+
+```ts
+export default config({
+    storage: {
+        my_image_storage: {
+            /* ... */
+        },
+    }
+    lists: {
+        images: { fields: { image({ storage: 'my_image_storage' }) } },
+        user: { fields: avatar: relationship({ ref: 'images' })}
+        blog: { fields: photos: relationship({ ref: 'images', many: true })}
+        /* ... */
+    },
+    /* ... */
+})
+```
+
+This allows mirroring of the old functionality, while allowing us to add the below feature/breaking change.
+
+##### Images and Files will now be deleted when deleted
+
+Before this change, if you uploaded a file or image, Keystone would never remove it from where it was stored. The inability to tidy up unused
+files or images was unwelcome. With the removal of `ref`, we can no remove things from the source, and this will be done by default.
```

Comment:
```suggestion
files or images was unwelcome. With the removal of `ref`, we can now remove things from the source, and this will be done by default.
```